packages/user_packages/iotcv2/iotcv2_toyota_sim.py

```diff
@@ -0,0 +1,98 @@
+                            cert_file=None, #self.config['edge_system_cert_file'],
+                            key_file=None #self.config['edge_system_key_file']
+                           )
+
+        tls_conf = TLSConf(self.config['cert_required'],
+                           self.config['tls_version'], None)
+
+        # Initialize DCC object with MQTT transport
+        self.iotcc = Iotcv2(
+                                      MqttDccComms(edge_system_name=edge_system.name,
+                                                   url=self.config['BrokerIP'],
+                                                   port=self.config['BrokerPort'],
+                                                   identity=identity,
+                                                   tls_conf=tls_conf,
+                                                   enable_authentication=True
+                                                   )
+                                      )
+        try:
+            self.iotcc_edge_system = self.iotcc.register(edge_system)
+            registry.register("iotc_v2_mqtt_wssl", self.iotcc)
+            registry.register("iotc_v2_mqtt_wssl_edge_system", self.iotcc_edge_system)
+        except RegistrationFailure:
+            print "EdgeSystem registration to IOTCV2 failed"
+            return
+        self.iotcc.set_properties(self.iotcc_edge_system, self.config['EdgeSystemPropList'])
+
+        # Register 19 ECU devices
+        self.reg_ecus = []
+        ecu_makes = ["Bosch", "Denso", "FujitsuTEN"]
+        ecu_models = [["BAEX1", "BWE2X", "BYTSE01", "BGGUTE22", "BFF1"], \
+                     ["DAEX1", "DWE2X", "DYTSE01", "DGGUTE22", "DFF1"], \
+                     ["FAEX1", "FWE2X", "FYTSE01", "FGGUTE22", "FFF1"]]
+        num_ecus = 19
+        try:
+            for i in range(0, num_ecus):
+                # Register device
+                log.info("*****Registeration Started for ECU %s" %(i))
+                ecu = SimulatedDevice(self.config['ECUNameList'][i] + '-' + os.environ['VIN'], "Toyota-ECU")
+                self.reg_ecu = self.iotcc.register(ecu)
+                self.reg_ecus.append(self.reg_ecu)
+                # create full ECU proplist
+                ecu_sn = '%0*d' % (12, (int(os.environ['MO_SN_START']) + i))
+                log.debug("ecu_sn:{0} PropList:{1}".format(ecu_sn, self.config['ECUPropList']))
+                prop_dict = {}
+                prop_dict.update({"Make": ecu_makes[i % 3]})
+                prop_dict.update({"Model": ecu_models[i % 3][i % 5]})
+                prop_dict.update({"SerialNumber": ecu_sn})
+                log.debug("new PropList:{0}".format(prop_dict))
+                self.iotcc.set_properties(self.reg_ecu, prop_dict)
+                self.iotcc.create_relationship(self.iotcc_edge_system, self.reg_ecu)
+        except Exception:
+            log.exception("Connected Device Registration and Metrics are Not Available")
```

Comment:
Log message seems incorrect because some of the 19 might be registered.

Reply:
modified